### PR TITLE
feat: add tooltip to the FormControl

### PR DESCRIPTION
**Pull request by boyarskiy**
https://autoricardo.atlassian.net/browse/VSST-849

## Motivation and context

Extend FormControl with a tooltip

## Before

N/A

## After

Screenshot 2022-12-15 at 18 58 58

## How to test

https://boyarskiy-vsst-849.components-pkg.branch.autoscout24.dev/?path=/story/components-forms-form-control--with-tooltip

### CODE REVIEW

File: src/components/formControl/index.tsx [hunk 1]
```diff
@@ -23,22 +28,35 @@ const FormControl: FC<PropsWithChildren<Props>> = ({
...
 
+  const renderLabel = (
+    <FormLabel htmlFor={id}>
+      {label}
+      {isRequired ? <>&nbsp;</> : null}
+    </FormLabel>
+  );
+
+  const renderLabelWithTooltip = (
+    <Flex justifyContent="start">
```
Comment (lkappeler) [inline src/components/formControl/index.tsx:43]: Stack might be more speaking in this case https://chakra-ui.com/docs/components/stack/usage#notes-on-stack-vs-flex

File: src/components/formControl/index.tsx [hunk 2]
```diff
@@ -23,22 +28,35 @@ const FormControl: FC<PropsWithChildren<Props>> = ({
   id,
   label,
   hint,
+  tooltip,
 }) => {
   const isInvalid = !!errorMessage;
 
+  const renderLabel = (
```
Comment (lkappeler) [inline src/components/formControl/index.tsx:35]: The `render` prefix feels a bit off for your variables as they do not render but only hold the JSX partial. In a react context I woul associate the `render` keyword with a function that renders the view.

**Review by lkappeler — COMMENTED** [review]
We might run into problems with this implementation as the tooltip will span the whole content width.  When looking at the designs I think the intention of the designer was to set a specific width.
image

File: src/components/tooltip/index.tsx [hunk 3]
```diff
@@ -2,9 +2,13 @@ import React, { FC } from 'react';
...
 
-const Tooltip: FC<TooltipProps> = ({ children, ...props }) => {
+type Props = {
+  children: React.ReactNode;
+} & Pick<TooltipProps, 'label' | 'placement' | 'maxWidth'>;
+
+const Tooltip: FC<Props> = ({ children, ...props }) => {
   return (
-    <ChakraTooltip hasArrow {...props}>
+    <ChakraTooltip hasArrow placement="auto" maxWidth="6xl" {...props}>
```
Comment (boyarskiy) [inline src/components/tooltip/index.tsx:11]: `maxWidth` confirmed by Sergiu

**Comment by boyarskiy** [comment]
> We might run into problems with this implementation as the tooltip will span the whole content width. When looking at the designs I think the intention of the designer was to set a specific width.

@lkappeler thanks, checked that with designer

File: src/components/tooltip/index.tsx (hunk 3 above)
Comment (boyarskiy) [inline src/components/tooltip/index.tsx:11]: `maxWidth` confirmed by Sergiu

**Review by lkappeler — APPROVED** [review]

**Pull request merged**